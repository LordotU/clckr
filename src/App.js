/**
 * @format
 * @flow
 */

import React, { useState, useEffect } from 'react';
import {
  SafeAreaView,
  StatusBar,
  Text,
  TouchableOpacity,
  TouchableWithoutFeedback,
  View,
} from 'react-native';
import SplashScreen from 'react-native-splash-screen';
import useStyles from './AppStyles';

export default function(): React$Node {
  useEffect(() => {
    SplashScreen.hide();
  }, []);

  const [count, setCount] = useState(0);
  const [highContrast, setHighContrast] = useState(false);

  const incrementCounter = () => setCount(count + 1);
  const resetCounter = () => setCount(0);
  const switchContrast = () => setHighContrast(!highContrast);

  const styles = useStyles(highContrast);

  return (
    <>
      <StatusBar barStyle={highContrast ? 'light-content' : 'dark-content'} />
      <SafeAreaView style={styles.outerView}>
        <TouchableWithoutFeedback
          accessible={true}
          accessibilityLabel="Increment counter"
          onPress={incrementCounter}>
          <View style={styles.innerView}>
            <Text style={styles.clickerCount} testID="counter">
              {count}
            </Text>
          </View>
        </TouchableWithoutFeedback>

        <TouchableOpacity
          accessible={true}
          accessibilityLabel="Reset counter"
          style={styles.button}
          onPress={resetCounter}>
          <Text style={styles.buttonInner}>Reset counter</Text>
        </TouchableOpacity>

        <TouchableOpacity
          accessible={true}
          accessibilityLabel="Switch contrast"
          style={styles.button}
          onPress={switchContrast}>
          <Text style={styles.buttonInner}>Switch contrast</Text>
        </TouchableOpacity>
      </SafeAreaView>
    </>
  );
}
