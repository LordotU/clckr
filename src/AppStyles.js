/**
 * @format
 * @flow
 */

import { StyleSheet } from 'react-native';
import { RFValue } from 'react-native-responsive-fontsize';

const FONT_SIZE_100 = 240;
const FONT_SIZE_100_DISPLAY_HEIGHT = 1242;

const defaults = {
  outerView: {
    flex: 1,
  },
  infoButton: {
    padding: 8,
    position: 'absolute',
    right: 24,
    top: 48,
    zIndex: 1,
  },
  infoIcon: {
    height: 32,
    width: 32,
  },
  innerView: {
    alignItems: 'center',
    flex: 1,
    justifyContent: 'center',
    width: '100%',
  },
  clickerCount: {
    fontFamily: 'WorkSans-Thin',
    fontSize: RFValue(FONT_SIZE_100, FONT_SIZE_100_DISPLAY_HEIGHT),
  },
  button: {
    paddingHorizontal: 32,
    paddingVertical: 12,
    width: '100%',
  },
  buttonInner: {
    borderColor: 'black',
    borderWidth: 0.5,
    color: 'black',
    fontFamily: 'WorkSans-Thin',
    fontSize: 24,
    padding: 8,
    textAlign: 'center',
    width: '100%',
  },
};

const contrast = {
  ...defaults,
  outerView: {
    ...defaults.outerView,
    backgroundColor: 'black',
  },
  clickerCount: {
    ...defaults.clickerCount,
    color: 'white',
  },
  buttonInner: {
    ...defaults.buttonInner,
    backgroundColor: 'black',
    borderColor: 'white',
    color: 'white',
  },
};

const stylesDefaults = StyleSheet.create(defaults);

const stylesContrast = StyleSheet.create(contrast);

export default function(useContrast: boolean) {
  return useContrast ? stylesContrast : stylesDefaults;
}
