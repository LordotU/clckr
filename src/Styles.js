/**
 * @format
 * @flow
 */

import { StyleSheet } from 'react-native';
import { RFValue } from 'react-native-responsive-fontsize';

const FONT_SIZE_100_DISPLAY_HEIGHT = 1242;

const defaults = {
  outerView: {
    flex: 1,
  },
  infoButton: {
    padding: 8,
    position: 'absolute',
    right: 24,
    top: 40,
    zIndex: 1,
  },
  infoIcon: {
    height: 32,
    width: 32,
  },
  innerView: {
    alignItems: 'center',
    flex: 1,
    justifyContent: 'center',
    width: '100%',
  },
  clickerCount: {
    color: 'black',
    fontFamily: 'WorkSans-Thin',
    fontSize: RFValue(240, FONT_SIZE_100_DISPLAY_HEIGHT),
  },
  button: {
    paddingHorizontal: 32,
    paddingVertical: 12,
    width: '100%',
  },
  buttonInner: {
    borderColor: 'black',
    borderWidth: 0.5,
    color: 'black',
    fontFamily: 'WorkSans-Thin',
    fontSize: 24,
    padding: 8,
    textAlign: 'center',
    width: '100%',
  },
  info: {
    alignItems: 'center',
    flex: 1,
    justifyContent: 'center',
  },
  outerInfoLink: {
    padding: 8,
  },
  infoLink: {
    fontFamily: 'WorkSans-Thin',
    fontSize: RFValue(48, FONT_SIZE_100_DISPLAY_HEIGHT),
    textDecorationLine: 'underline',
  },
  infoVersion: {
    fontFamily: 'WorkSans-Thin',
    fontSize: RFValue(40, FONT_SIZE_100_DISPLAY_HEIGHT),
  },
};

const contrast = {
  ...defaults,
  outerView: {
    ...defaults.outerView,
    backgroundColor: 'black',
  },
  clickerCount: {
    ...defaults.clickerCount,
    color: 'white',
  },
  buttonInner: {
    ...defaults.buttonInner,
    backgroundColor: 'black',
    borderColor: 'white',
    color: 'white',
  },
  infoLink: {
    ...defaults.infoLink,
    color: 'white',
  },
  infoVersion: {
    ...defaults.infoVersion,
    color: 'white',
  },
};

const stylesDefaults = StyleSheet.create(defaults);

const stylesContrast = StyleSheet.create(contrast);

export default function(useContrast: boolean = false) {
  return useContrast ? stylesContrast : stylesDefaults;
}
