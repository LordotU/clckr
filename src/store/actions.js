/**
 * @format
 * @flow
 */

type ExtractReturn<Fn> = $Call<<T>((...Iterable<any>) => T) => T, Fn>;

export const actionTypes = { SWITCH_CONTRAST: 'SWITCH_CONTRAST' };

export const switchContrast = (highContrast: boolean) => ({
  type: actionTypes.SWITCH_CONTRAST,
  payload: highContrast,
});

export type Action = ExtractReturn<typeof switchContrast>;
