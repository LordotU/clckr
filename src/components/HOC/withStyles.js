/* eslint-disable react-hooks/rules-of-hooks */

/**
  @flow
 * @format
 */

import React, { useContext } from 'react';

import { context } from '../../store';
import useStyles from '../../Styles';

export default function(Component) {
  return function withStyles(props) {
    const {
      state: { highContrast },
    } = useContext(context);

    const styles = useStyles(highContrast);

    return <Component styles={styles} {...props} />;
  };
}
