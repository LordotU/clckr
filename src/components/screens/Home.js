/**
 * @format
 * @flow
 */

import React, { useContext, useState } from 'react';
import {
  Image,
  Text,
  TouchableOpacity,
  TouchableWithoutFeedback,
  View,
} from 'react-native';

import { context } from '../../store';
import { switchContrast as setHighContrast } from '../../store/actions';

const InfoIconDark = require('../../../assets/images/icon-info-dark.png');
const InfoIconLight = require('../../../assets/images/icon-info-light.png');

export default function({ navigation, styles }): React$Node {
  const [count, setCount] = useState(0);

  const {
    state: { highContrast },
    dispatch,
  } = useContext(context);

  const incrementCounter = () => setCount(count + 1);
  const resetCounter = () => setCount(0);
  const switchContrast = () => dispatch(setHighContrast(!highContrast));

  return (
    <>
      <TouchableOpacity
        accessible={true}
        accessibilityLabel="Show info"
        style={styles.infoButton}
        onPress={() => navigation.navigate('Info')}>
        <Image
          source={highContrast ? InfoIconLight : InfoIconDark}
          style={styles.infoIcon}
        />
      </TouchableOpacity>

      <TouchableWithoutFeedback
        accessible={true}
        accessibilityLabel="Increment counter"
        onPress={incrementCounter}>
        <View style={styles.innerView}>
          <Text style={styles.clickerCount} testID="counterText">
            {count}
          </Text>
        </View>
      </TouchableWithoutFeedback>

      <TouchableOpacity
        accessible={true}
        accessibilityLabel="Reset counter"
        style={styles.button}
        onPress={resetCounter}>
        <Text style={styles.buttonInner}>Reset counter</Text>
      </TouchableOpacity>

      <TouchableOpacity
        accessible={true}
        accessibilityLabel="Switch contrast"
        style={styles.button}
        onPress={switchContrast}>
        <Text style={styles.buttonInner}>Switch contrast</Text>
      </TouchableOpacity>
    </>
  );
}
