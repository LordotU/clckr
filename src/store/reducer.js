/**
 * @format
 * @flow
 */

import { createContext } from 'react';
import { actionTypes } from './actions';
import type { Action } from './actions';

export const defaultState = {
  highContrast: false,
};

export const context = createContext<Object>(defaultState);

export const reducer = (
  state: typeof defaultState,
  { type, payload }: Action,
) => {
  switch (type) {
    case actionTypes.SWITCH_CONTRAST:
      return { ...state, highContrast: payload };
    default:
      return state;
  }
};
