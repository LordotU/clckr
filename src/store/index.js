/**
 * @format
 * @flow
 */

import React, { useReducer } from 'react';
import { context, defaultState, reducer } from './reducer';

const { Provider }: { Provider: Object } = context;

const StateProvider = ({ children }: { children: React$Node }) => {
  const [state, dispatch] = useReducer(reducer, defaultState);

  return <Provider value={{ state, dispatch }}>{children}</Provider>;
};

export { context, StateProvider };
