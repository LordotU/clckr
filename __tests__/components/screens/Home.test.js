/**
 * @format
 */

import React from 'react';
import { Image } from 'react-native';
import { cleanup, fireEvent, render } from 'react-native-testing-library';

import Home from '../../../src/components/screens/Home';
import withStyles from '../../../src/components/HOC/withStyles';
import { StateProvider } from '../../../src/store';

const HomeWithStyles = withStyles(Home);

const getHomeScreen = () => (
  <StateProvider>
    <HomeWithStyles />
  </StateProvider>
);

afterEach(cleanup);

it('renders correctly', () => {
  const snapshot = render(getHomeScreen()).toJSON();
  expect(snapshot).toMatchSnapshot();
});

it('home screen works correctly', () => {
  const screen = render(getHomeScreen());

  const infoImage = screen.getByType(Image);
  expect(infoImage.props.source.testUri).toMatch('icon-info-dark');

  const counterText = screen.getByTestId('counterText');
  expect(counterText.props.style.color).toBe('black');

  const counterBtn = screen.getByA11yLabel('Increment counter');
  fireEvent.press(counterBtn);
  fireEvent.press(counterBtn);
  expect(counterText.props.children).toBe(2);

  const resetBtn = screen.getByA11yLabel('Reset counter');
  fireEvent.press(resetBtn);
  expect(counterText.props.children).toBe(0);

  const contrastBtn = screen.getByA11yLabel('Switch contrast');
  fireEvent.press(contrastBtn);
  expect(counterText.props.style.color).toBe('white');
  expect(infoImage.props.source.testUri).toMatch('icon-info-light');
  fireEvent.press(contrastBtn);
  expect(counterText.props.style.color).toBe('black');
  expect(infoImage.props.source.testUri).toMatch('icon-info-dark');
});
