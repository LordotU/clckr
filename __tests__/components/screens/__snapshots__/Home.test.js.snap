// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <View
    accessibilityLabel="Show info"
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
        "padding": 8,
        "position": "absolute",
        "right": 24,
        "top": 40,
        "zIndex": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/images/icon-info-dark.png",
        }
      }
      style={
        Object {
          "height": 32,
          "width": 32,
        }
      }
    />
  </View>,
  <View
    accessibilityLabel="Increment counter"
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontFamily": "WorkSans-Thin",
          "fontSize": 258,
        }
      }
      testID="counterText"
    >
      0
    </Text>
  </View>,
  <View
    accessibilityLabel="Reset counter"
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
        "paddingHorizontal": 32,
        "paddingVertical": 12,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "borderColor": "black",
          "borderWidth": 0.5,
          "color": "black",
          "fontFamily": "WorkSans-Thin",
          "fontSize": 24,
          "padding": 8,
          "textAlign": "center",
          "width": "100%",
        }
      }
    >
      Reset counter
    </Text>
  </View>,
  <View
    accessibilityLabel="Switch contrast"
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
        "paddingHorizontal": 32,
        "paddingVertical": 12,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "borderColor": "black",
          "borderWidth": 0.5,
          "color": "black",
          "fontFamily": "WorkSans-Thin",
          "fontSize": 24,
          "padding": 8,
          "textAlign": "center",
          "width": "100%",
        }
      }
    >
      Switch contrast
    </Text>
  </View>,
]
`;
